systemcontractindex/stakingindex/index.go

```diff
@@ -41,28 +45,57 @@ type (
 		TotalBucketCount(height uint64) (uint64, error)
 		PutBlock(ctx context.Context, blk *block.Block) error
 	}
+	stakingEventHandler interface {
+		HandleStakedEvent(event *abiutil.EventParam) error
+		HandleLockedEvent(event *abiutil.EventParam) error
+		HandleUnlockedEvent(event *abiutil.EventParam) error
+		HandleUnstakedEvent(event *abiutil.EventParam) error
+		HandleDelegateChangedEvent(event *abiutil.EventParam) error
+		HandleWithdrawalEvent(event *abiutil.EventParam) error
+		HandleTransferEvent(event *abiutil.EventParam) error
+		HandleMergedEvent(event *abiutil.EventParam) error
+		HandleBucketExpandedEvent(event *abiutil.EventParam) error
+		HandleDonatedEvent(event *abiutil.EventParam) error
+		Finalize() (batch.KVStoreBatch, *cache)
```

Comment:
nit: this is an internal interface, can start with lowercase like `handleStakedEvent`, then no need to change funcs names above